### PR TITLE
ENH: QIF regression

### PR DESCRIPTION
**Pull request by kshedden**

### CODE REVIEW

File: statsmodels/genmod/qif.py [hunk 1]
```diff
@@ -0,0 +1,504 @@
...
+        The independent variables of the regression.
+    groups : array-like
+        Labels indicating which group each observation belongs to.
+        Observations in different groups should be independent.
+    family : genmod family
+        An instance of a GLM family.
+    cov_struct : QIFCovariance instance
+        An instance of a QIFCovariance.
+
+    Returns
```
Comment (bashtage) [inline statsmodels/genmod/qif.py]: Not sure that __init__ returns anything here, other than the object.

File: statsmodels/genmod/qif.py [hunk 2]
```diff
@@ -0,0 +1,504 @@
...
+
+class QIFAutoregressive(QIFCovariance):
+    """
+    Autoregressive working covariance for QIF regression.
+    """
+
+    def __init__(self):
+        self.num_terms = 3
+
+    def mat(self, dim, term):
```
Comment (josef-pkt) [inline statsmodels/genmod/qif.py]: what is `mat`? Is there a more specific, descriptive name?

File: statsmodels/genmod/qif.py [hunk 3]
```diff
@@ -0,0 +1,506 @@
...
+
+    References
+    ----------
+    A. Qu, B. Lindsay, B. Li (2000).  Improving Generalized Estimating
+    Equations using Quadratic Inference Functions, Biometrika 87:4.
+    www.jstor.org/stable/2673612
+    """
+
+    def __init__(self, endog, exog, groups, family=None,
+                 cov_struct=None, missing='none', **kwargs):
```
Comment (jbrockmendel) [inline statsmodels/genmod/qif.py]: what kwargs are expected?  Can these be made explicit?

Comment (jbrockmendel) [reply statsmodels/genmod/qif.py]: Never mind, didn't see this subclasses Model.

File: statsmodels/genmod/tests/test_qif.py [hunk 4]
```diff
@@ -0,0 +1,111 @@
+import numpy as np
+from statsmodels.genmod.qif import (QIF, QIFIndependence, QIFExchangeable,
+     QIFAutoregressive)
+from statsmodels.tools.numdiff import approx_fprime
+from numpy.testing import assert_allclose
+import pandas as pd
+import pytest
```
Comment (jbrockmendel) [inline statsmodels/genmod/tests/test_qif.py]: can you collect 3rd party imports before statsmodels imports

**Pull request merged**